modules/client/src/main/java/org/jboss/wsf/stack/cxf/client/configuration/ElytronClientConfig.java

```diff
@@ -0,0 +1,123 @@
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.jboss.wsf.stack.cxf.client.configuration;
+
+import org.jboss.wsf.spi.security.WildFlyClientConfigProvider;
+
+import javax.net.ssl.SSLContext;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+/**
+ * Class for storing of authentication information obtained from Elytron client configuration
+ *
+ * @author dvilkola@redhat.com
+ * @since 24-Jul-2019
+ */
+class ElytronClientConfig {
+
+   private static String username;
+   private static String password;
+   private static SSLContext sslContext;
+   private static HttpMechanismType httpMechanism;
+   private static WsSecurityType wsSecurityType;
+
+   enum HttpMechanismType {
+      BASIC("BASIC");
+
+      private String text;
+
+      HttpMechanismType(String text) {
+         this.text = text;
+      }
+
+      public String getText() {
+         return this.text;
+      }
+
+      public static HttpMechanismType fromString(String text) {
+         for (HttpMechanismType httpMechanismType : HttpMechanismType.values()) {
+            if (httpMechanismType.text.equalsIgnoreCase(text)) {
+               return httpMechanismType;
+            }
+         }
+         return null;
+      }
+   }
+
+   enum WsSecurityType {
+      USERNAME_TOKEN("UsernameToken");
+
+      private String text;
+
+      WsSecurityType(String text) {
+         this.text = text;
+      }
+
+      public String getText() {
+         return this.text;
+      }
+
+      public static WsSecurityType fromString(String text) {
+         for (WsSecurityType wsSecurityType : WsSecurityType.values()) {
+            if (wsSecurityType.text.equalsIgnoreCase(text)) {
+               return wsSecurityType;
+            }
+         }
+         return null;
+      }
+   }
+
+   static String getUsername() {
+      return username;
+   }
+
+   static String getPassword() {
+      return password;
+   }
+
+   static SSLContext getSslContext() {
+      return sslContext;
+   }
+
+   static HttpMechanismType getHttpMechanism() {
+      return httpMechanism;
+   }
+
+   static WsSecurityType getWsSecurityType() {
+      return wsSecurityType;
+   }
+
+   static void setAuthConfiguration(WildFlyClientConfigProvider wildflyConfigProvider, String endpointAddress) {
+      URI endpointURI = null;
+      try {
+         endpointURI = new URI(endpointAddress);
+      } catch (URISyntaxException e) {
+         e.printStackTrace();    // TODO throw some uri parsing exception
```

Comment:
What about this TODO? I guess client won't be used only as standalone application but can run at container environment so we need exception here, right?

Reply:
Fixed.